sys/usb/usbus/cdc/acm/cdc_acm_stdio.c

```diff
@@ -35,6 +36,44 @@ static uint8_t _cdc_tx_buf_mem[USBUS_CDC_ACM_STDIO_BUF_SIZE];
 static uint8_t _cdc_rx_buf_mem[USBUS_CDC_ACM_STDIO_BUF_SIZE];
 static isrpipe_t _cdc_stdio_isrpipe = ISRPIPE_INIT(_cdc_rx_buf_mem);
 
+#ifdef MODULE_ARDUINO_BOOTLOADER
+
+#ifndef RESET_IN_BOOTLOADER_TRIGGER_BAUDRATE
+#define RESET_IN_BOOTLOADER_TRIGGER_BAUDRATE    (1200U)
+#endif
+
+#ifndef RESET_IN_APPLICATION_TRIGGER_BAUDRATE
+#define RESET_IN_APPLICATION_TRIGGER_BAUDRATE   (600U)
+#endif
+
+extern void reset_in_bootloader(void);
+extern void reset_in_application(void);
+
+static int _coding_cb(usbus_cdcacm_device_t *cdcacm, uint32_t baud, uint8_t bits,
+                      uint8_t parity, uint8_t stop)
+{
+    (void)cdcacm;
+    (void)bits;
+    (void)parity;
+    (void)stop;
+    switch (baud) {
+        case RESET_IN_BOOTLOADER_TRIGGER_BAUDRATE:
+            LOG_DEBUG("[cdc-acm-stdio] reset in bootloader");
+            reset_in_bootloader();
+            break;
+        case RESET_IN_APPLICATION_TRIGGER_BAUDRATE:
+            LOG_DEBUG("[cdc-acm-stdio] reset in application");
+            reset_in_application();
+            break;
+        default:
+            (void)baud;
+            break;
+    }
+
+    return 0;
+}
+#endif /* MODULE_ARDUINO_BOOTLOADER */
+
```

Comment:
Honestly I'm not very happy with this code in the common cdc_acm_stdio file. Is there a common Arduino board file where we could move this? Maybe the new `reset.c` file from above as this is also reset-related?